SpatialGDK/Source/Private/Interop/SpatialSender.cpp

```diff
@@ -112,11 +118,11 @@ Worker_RequestId USpatialSender::CreateEntity(const FString& ClientWorkerId, con
 
 	TArray<Worker_ComponentData> ComponentDatas;
 	ComponentDatas.Add(improbable::Position(improbable::Coordinates::FromFVector(Channel->GetActorSpatialPosition(Actor))).CreatePositionData());
-	ComponentDatas.Add(improbable::Metadata(EntityType).CreateMetadataData());
+	ComponentDatas.Add(improbable::Metadata(Channel->Actor->GetClass()->GetPathName()).CreateMetadataData());
```

Comment:
Why the change?

Reply:
Do we need null checks here?

Reply:
Change is because `EntityType` was being passed as a parameter when it didn't need to be. No null checks are needed since you have the actor at this point in every case.